sympy/integrals/transforms.py

```diff
@@ -1005,14 +1007,45 @@ def repl(ex, *args):
     expr = repl(expr, Unequality, replue)
     return S(expr)
 
+def expand_dirac_delta(expr):
+    """
+    Expand an expression involving DiractDelta to get it as a linear
+    combination of DiracDelta functions.
+    """
+    return _lin_eq2dict(expr, expr.atoms(DiracDelta))
 
 @_noconds
 def _laplace_transform(f, t, s_, simplify=True):
     """ The backend function for Laplace transforms. """
     from sympy import (re, Max, exp, pi, Min, periodic_argument as arg_,
-                       arg, cos, Wild, symbols, polar_lift)
+                       arg, cos, Wild, symbols, polar_lift, Add)
     s = Dummy('s')
-    F = integrate(exp(-s*t) * f, (t, 0, oo))
+    a = Wild('a', exclude=[t])
+    deltazero = 0
+    try:
+        integratable, deltadict = expand_dirac_delta(f)
+    except PolyNonlinearError:
+        raise IntegralTransformError(
+        'Laplace', f, 'could not expand DiracDelta expressions')
+    for dd in deltadict.keys():
+        if len(dd.args)>1 and dd.args[1]>0:
+            raise IntegralTransformError(
+                'Laplace', f, 'derivatives of DiracDelta not implemented')
+        argzero = dd.args[0].subs(t,0).is_zero
+        if argzero is True:
+            p = dd.args[0].match(a*t)
+            if p:
+                deltazero = Add(deltazero,(deltadict[dd].subs(t,0)/p[a]))
+            else:
+                raise IntegralTransformError('Laplace', f, \
+                                             'argument of DiracDelta is not \
+                                              a 1st-order polynomial in t')
+        elif argzero is False:
+            integratable = Add(integratable,dd*deltadict[dd])
```

Comment:
Calling `Add` repeatedly in a loop is `O(N**2)`. You should collect the terms in a list and the call `Add(*terms)` once. 

Also standard formatting puts a space after a comma.

Reply:
And I have again learned something ;)))))